### DISCUSSION TITLE
x-interpolate

### DISCUSSION THREAD

**Discussion by earthboundkid** [alpinejs/alpine#1708] [3. Feature ideas]
**Problem**: x-text is great for replace all text, but a pain when you only want to replace a subset:

```html
<h1 x-data="{ n: 1 }" @click="n++">Hello, #<span x-text="n"></span></h1>
```

**Solution**: x-interpolate

```html
<h1 x-data="{ n: 1 }" @click="n++" x-interpolate>Hello, #${n}</h1>
```

Code for the plugin:

```javascript
Alpine.directive("interpolate", (el, _, { effect, evaluateLater }) => {
  let evaluate = evaluateLater("`" + el.innerText + "`");

  effect(() => {
    evaluate((value) => {
      el.textContent = value;
    });
  });
});
```

What do y'all think?

CodePen: https://codepen.io/carlmjohnson/pen/QWvbeLZ

**Comment by earthboundkid** [comment]
There is probably a better name than "interpolate". It could maybe be an option on x-text, like 

```html
<h1 x-data="{ n: 1 }" @click="n++" x-text.replace>Hello, #${n}</h1>
```

**Comment by KevinBatdorf** [comment]
I don't think `x-text.replace` is better but I like the idea. `x-inline-template` would probably be the more obvious naming

**Reply by earthboundkid** [reply]
I like the name `x-template`.

**Reply by KevinBatdorf** [reply]
Vue uses `inline-template` but yeah even just `x-template` sounds like a good choice for naming.

**Reply by earthboundkid** [reply]
This is different from Vue because, for example, it overwrites any tags inside of the element, whereas Vue hydrates them. I think it's good to have a distinct name, so people don't expect the same behavior.

**Reply by KevinBatdorf** [reply]
Good point. 

By the way, I'd probably attach a codepen showing it off in your original post so others can see it more quickly.

**Reply by earthboundkid** [reply]
https://codepen.io/carlmjohnson/pen/QWvbeLZ

**Comment by earthboundkid** [comment]
One more thought. ``x-text="`Hello, ${name}`"`` is awkward to write because of the quotes in quotes. What if `x-template` looks at whether it has an argument or not, and if there's no argument, it uses innerText, and if there is, it uses the expression. That way you can write the pre-init fallback text in the HTML and then have it overwritten once Alpine loads.

https://codepen.io/carlmjohnson/pen/ZEKbzNr
